Client/Client.Qt/ChatWindow.cpp

```diff
@@ -8,14 +8,26 @@ ChatWindow::ChatWindow(QWidget* parent) : QWidget(parent)
     setContentsMargins(0, 0, 0, 0);
     _mainLayout        = std::make_unique<QSplitter>(this);
     _channelListWidget = std::make_unique<ChannelListWidget>();
-    _chat              = std::make_unique<ChatWidget>();
+    _chatSwitchWidget  = std::make_unique<QStackedWidget>();
     _mainLayout->setOrientation(Qt::Horizontal);
     _mainLayout->setChildrenCollapsible(false);
     _mainLayout->setContentsMargins(0, 0, 0, 0);
     _mainLayout->addWidget(_channelListWidget.get());
-    _mainLayout->addWidget(_chat.get());
-    _mainLayout->setStretchFactor(0, 20);
-    _mainLayout->setStretchFactor(1, 80);
+    _mainLayout->addWidget(_chatSwitchWidget.get());
+
+    connect(_channelListWidget->getChannelList().get(), &ListWidget::itemPressed,
+            [&](){
+            if(_chatSwitchWidget->count() == _channelListWidget->getChannelList()->currentRow())
+            {
+                _chatSwitchWidget->insertWidget(_channelListWidget->getChannelList()->currentRow(), new ChatWidget());
+                _mainLayout->setStretchFactor(1, 80);
```

Comment:
Do not change streatch factor on every single chat. Set it once:
_mainLayout->setStretchFactor(0, 20);
_mainLayout->setStretchFactor(1, 80);
This is because user can drag splitter to change it manually

Reply:
Doesn't it change 1 time?
Because if I change 1 chat and add the 2nd, changes will remain

Reply:
You change user's stretch factor on chat change. It mustn't happen.